File: 0551-student-attendance-record-i/0551-student-attendance-record-i.java
class Solution {
    public boolean checkRecord(String s) {
        int n = s.length();

        int absCount = 0;
        int lateCount = 0;

        for(int i=0; i<n;i++){
            char ch = s.charAt(i);

            if(ch=='A') {
                absCount++;
                if(absCount>=2) return false;
            }
            if (ch=='L'){
                lateCount++;
                if(lateCount>=3) return false;
            }
            else lateCount = 0;
        }
        return true;

    }
}

File: 0551-student-attendance-record-i/0551-student-attendance-record-i.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        n = len(s)
        absCount =  0
        lateCount = 0
        for i in range(n):
            ch = s[i]
            if ch=='A' :
                absCount+=1
                if(absCount>=2) : return False
            if ch=='L':
                lateCount+=1
                if(lateCount>=3): return False
            
            else: lateCount = 0
        return True
